llarp/router/router.cpp

```diff
@@ -811,29 +832,30 @@ namespace llarp
   }
 
   void
-  Router::SetRouterWhitelist(const std::vector<RouterID>& routers)
+  Router::SetRouterWhitelist(const std::vector<RouterID> routers)
   {
     _rcLookupHandler.SetRouterWhitelist(routers);
   }
 
   bool
-  Router::StartJsonRpc()
+  Router::StartRpcServer()
   {
-    if (_running || _stopping)
-      return false;
-
     if (enableRPCServer)
     {
-      if (rpcBindAddr.isEmpty())
+      if (rpcBindAddr.empty())
       {
         rpcBindAddr = DefaultRPCBindAddr;
       }
-      rpcServer = std::make_unique<rpc::Server>(this);
-      if (not rpcServer->Start(rpcBindAddr))
+
+      // older configs just specify an ip/port tuple so check for that and prepend tcp:// in that
+      // case
+      if (rpcBindAddr.find("tcp://") == std::string::npos
+          and rpcBindAddr.find("ipc://") == std::string::npos)
```

Comment:
I would make this a little more general by just looking for the `://`: `if (rpcBindAddr.find("://") == std::string::npos)` because there are some other address types like `curve://` or `ipc+curve://` which can be used with LMQ if you want to connect to a curve-encrypted listener.

Reply:
+1